src/commands/lambda/instrument.ts

```diff
@@ -127,6 +143,46 @@ export class InstrumentCommand extends Command {
     return value ? value.toLowerCase() === 'true' : configValue ? configValue.toLowerCase() === 'true' : fallback
   }
 
+  private async getCurrentGitStatus() {
+    const simpleGit = await newSimpleGit()
+    const gitCommitInfo = await getCommitInfo(simpleGit, this.context.stdout)
+    if (gitCommitInfo === undefined) {
+      return 1
+    }
+    const status = await simpleGit.status()
+
+    return {isClean: status.isClean(), ahead: status.ahead, files: status.files, hash: gitCommitInfo?.hash}
+  }
+
+  private async getGitDataAndUpload(settings: InstrumentationSettings) {
+    const currentStatus = await this.getCurrentGitStatus()
+
+    if (currentStatus === 1) {
+      return 1
```

Comment:
Instead of returning 0 or 1 from a function to indicate whether there was a problem, it would be more idiomatic in TypeScript to throw an error if something goes wrong, and return nothing if all is well.

Reply:
Agreed, but it looks like this file actually has 15 cases of `return 1` and 3 cases of `return 0`. I think it might make more sense to keep this convention, at least for now. Do you agree?

Reply:
I don't really think so -- I don't think returning 0 and 1 from functions is idiomatic in TypeScript, we should fix it or at the very least not make the problem worse.